src/date-adapters/luxon/index.ts

```diff
@@ -0,0 +1,187 @@
+import { DateTime } from 'luxon';
+import { DateAdapter } from '../date-adapter';
+
+export function adapterFactory(luxon): DateAdapter {
+  /* istanbul ignore next */
+  function coerceDateTime(date: Date | number): DateTime {
+    return typeof date === 'number'
+      ? luxon.DateTime.fromMillis(date)
+      : luxon.DateTime.fromJSDate(date);
+  }
+
+  function addDays(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ days: amount }).toJSDate();
+  }
+
+  function addHours(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ hours: amount }).toJSDate();
+  }
+
+  function addMinutes(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ minutes: amount }).toJSDate();
+  }
+
+  function addSeconds(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ seconds: amount }).toJSDate();
+  }
+
+  function differenceInDays(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): number {
+    return Math.trunc(
+      coerceDateTime(dateLeft).diff(coerceDateTime(dateRight)).as('days')
+    );
+  }
+
+  function differenceInMinutes(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): number {
+    return Math.trunc(
+      coerceDateTime(dateLeft).diff(coerceDateTime(dateRight)).as('minutes')
+    );
+  }
+
+  function differenceInSeconds(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): number {
+    return Math.trunc(
+      coerceDateTime(dateLeft).diff(coerceDateTime(dateRight)).as('seconds')
+    );
+  }
+
+  function endOfDay(date: Date | number): Date {
+    return coerceDateTime(date).endOf('day').toJSDate();
+  }
+
+  function endOfMonth(date: Date | number): Date {
+    return coerceDateTime(date).endOf('month').toJSDate();
+  }
+
+  function endOfWeek(
+    date: Date | number,
+    options?: {
+      weekStartsOn?: number;
+    }
+  ): Date {
+    return coerceDateTime(date)
+      .endOf('week', { useLocaleWeeks: true })
+      .plus({ days: /* istanbul ignore next */ options?.weekStartsOn ?? 0 })
+      .toJSDate();
+  }
+
+  function getDay(date: Date | number): number {
+    return coerceDateTime(date).weekday % 7;
```

Comment:
How come we need to use ` % 7`? Does this API not return an integer?

Reply:
Luxon uses the ISO 8601 weekday numbers, which start with 1 on Monday and end with 7 on Sunday. However, our enum, like Moment.js and others, defines 0 as Sunday, 1 as Monday, and so on.

Reply:
Ahh I see now, can we add a comment explaining that?

Reply:
Done